src/test/java/ComplicatedOperationServiceTest.java

```diff
@@ -0,0 +1,392 @@
+import calculator.model.Input;
+import calculator.service.BasicOperationService;
+import calculator.service.ComplicatedOperationService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.internal.matchers.Null;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.hamcrest.Matchers.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ComplicatedOperationServiceTest {
+
+    private ComplicatedOperationService cOS;
+    //Inputs for sum, subtraction, multiplication, and division
+    private Input posAPosB;
+    private Input posANegB;
+    private Input negAPosB;
+    private Input negANegB;
+    private Input posAZeroB;
+    private Input zeroAPosB;
+    private Input zeroAZeroB;
+    private Input posANullB;
+    private Input nullAPosB;
+
+    private Input nullInput;
+    private Input nullANullB;
+
+    private List<Double> posPos;
+    private List<Double> posNeg;
+    private List<Double> negPos;
+    private List<Double> negNeg;
+    private List<Double> zeroPos;
+
+
+    @Mock
+    private BasicOperationService bOS;
+
+    @Before
+    public void setup() {
+        //Inputs setup
+        posAPosB = new Input(1,1);
+        posANegB = new Input(1, -1);
+        negAPosB = new Input(-1, 1);
+        negANegB = new Input(-1, -1);
+        posAZeroB = new Input(1, 0);
+        zeroAPosB = new Input(0, 1);
+        zeroAZeroB = new Input(0, 0);
+        posANullB = new Input(1, null);
+        nullAPosB = new Input(null, 1);
+        nullANullB = new Input(null, null);
+        nullInput = null;
+
+        cOS = new ComplicatedOperationService();
+        cOS.setBasicOperationService(bOS);
+
```

Comment:
for someone who doesn't know the code this much, it would be helpful to give the name which explains what it is, even if it feels it's too long sometimes, eg. basicService / complicatedService would sound better cause you already know that it's service and it should contains some methods